kafka-plugins/src/main/java/co/cask/hydrator/plugin/realtime/KafkaProducer.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.realtime;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.realtime.DataWriter;
+import co.cask.cdap.etl.api.realtime.RealtimeContext;
+import co.cask.cdap.etl.api.realtime.RealtimeSink;
+import co.cask.cdap.etl.common.StructuredRecordStringConverter;
+import com.google.common.collect.Lists;
+import kafka.producer.ProducerConfig;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.StringWriter;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Implementation of Kafka Realtime Producer Hydrator plugin. 
+ * 
+ * The producer has the capability to transform a {@link StructuredRecord}
+ * into a CSV or JSON record and push it on to one or more Kafka topics. 
+ * Producer can use one of the fields in the input records to partition the 
+ * data. It can also be configured to operate in sync or async mode.  
+ */
+@Plugin(type = "realtimesink")
+@Name("KafkaProducer")
+@Description("Real-time Kafka producer")
+public class KafkaProducer extends RealtimeSink<StructuredRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaProducer.class);
+
+  // Configuration for the plugin.
+  private final Config sconfig;
+  
+  // Static constants for configuring Kafka producer. 
+  private static final String BROKER_LIST = "bootstrap.servers";
```

Comment:
Kafka producer supports quite a number of configuration. Shall we have an additional config map that can be provided by the user to specify any producer config?

http://kafka.apache.org/documentation.html#producerconfigs